compiler/rustc_data_structures/src/thin_vec.rs

```diff
@@ -46,6 +56,102 @@ impl<T> ::std::ops::DerefMut for ThinVec<T> {
     }
 }
 
+impl<T> FromIterator<T> for ThinVec<T> {
+    fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
+        let mut iter = iter.into_iter();
+
+        match iter.next() {
+            None => ThinVec(None),
+            Some(first) => {
+                // + 1 for the first element
+                let mut vec = Vec::with_capacity(iter.size_hint().0 + 1);
+
+                vec.push(first);
+                for next in iter {
```

Comment:
This could be `vec.extend(iter);`

Reply:
:+1: (by the way, that code is from #83821)

Reply:
Done in 530c7ee36e4.